packages/app-store/routing-forms/pages/route-builder/[...appPages].tsx

```diff
@@ -553,11 +553,18 @@ export default function RouteBuilder({
     <SingleForm
       form={form}
       appUrl={appUrl}
-      Page={({ hookForm, form }) => (
-        <div className="route-config">
-          <Routes hookForm={hookForm} appUrl={appUrl} form={form} />
-        </div>
-      )}
+      Page={({ hookForm, form }) => {
+        // If hookForm hasn't been initialized, don't render anything
+        // This is important here because some states get initialized which aren't reset when the hookForm is reset with the form values and they don't get the updated values
+        if (!hookForm.getValues().id) {
```

Comment:
The code was flaky earlier, but no idea why it's consistently failing now but it's the right fix.